GGJ2020/Assets/Scripts/Collector.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using System.Linq;
+
+public class Collector : MonoBehaviour
+{
+    [Tooltip("How close the mouse has to be to the collectable for it to be clickable")]
+    public float mouseDistance = .5f;
+
+    [Tooltip("How close this collector has to be to the collectable for it to be clickable")]
+    public float collectionDistance = 2f;
+
+    [Tooltip("Needed to calculate the mouses cursors world position")]
+    public float distanceOfCameraFromGround = 10f;
+
+    [Tooltip("Input.GetButtonUp to use for clicking")]
+    public string ClickButton = "Fire1";
+
+    public Collectable lastHovering;
+    public Collectable[] allItems;
+
+    private void Awake()
+    {
+        allItems = FindObjectsOfType<Collectable>();
+    }
+
+    // void Start() { }
+
+    void Update()
+    {
+        updateItemClickability();
+
+        if (lastHovering != null && Input.GetButtonUp(ClickButton))
+        {
+            lastHovering.ClickItGood();
+        }
+    }
+
+    void updateItemClickability()
+    {
+        float distanceToPlayer;
+        List<Collectable> inCollectorRange = new List<Collectable>();
+
+        foreach (var collectable in allItems)
+        {
+            distanceToPlayer =
+                Vector3.Distance(this.transform.position, collectable.transform.position);
+
+            // highlight objects collecatable by player
+            if (distanceToPlayer < collectionDistance
+                // TODO: && check is in Line of Sight
+                )
+            {
+                collectable.SetClickability(Collectable.ClickabilityEnum.Clickable);
+                inCollectorRange.Add(collectable);
+            }
+            else
+            {
+                collectable.SetClickability(Collectable.ClickabilityEnum.OutOfRange);
+            }
+        }
+
+        if (inCollectorRange == null || inCollectorRange.Count <= 0)
+            return;
+
+        // determine players mouse cursor position in world
+        var mouseInWorld = Input.mousePosition;
+        mouseInWorld.z = distanceOfCameraFromGround;
+        mouseInWorld = Camera.main.ScreenToWorldPoint(mouseInWorld);
```

Comment:
This is really inefficient, I would rather see you use a circle cast for this.

Reply:
Fixed in new commit.